### ISSUE TITLE
Failed to establish a new connection: [Errno -4] Non-recoverable error

### ISSUE THREAD

**Issue by djosborne** [kennethreitz/requests#3910]
Hey there, I'm using requests (2.9.1) to make an https request to a kubernetes apiserver but am hitting an error and would appreciate some debugging tips on how to figure out what is causing it. 

The error:

```
Traceback (most recent call last):
  File "<string>", line 304, in _manage_resource
  File "<string>", line 393, in _sync_resources
  File "<string>", line 445, in _api_get
  File "site-packages/requests/sessions.py", line 480, in get
  File "site-packages/requests/sessions.py", line 468, in request
  File "site-packages/requests/sessions.py", line 576, in send
  File "site-packages/requests/adapters.py", line 437, in send
ConnectionError: HTTPSConnectionPool(host='<hostname redacted>', port=8443): Max retries exceed
ed with url: /api/v1/namespaces (Caused by NewConnectionError('<requests.packages.urllib3.connection.
VerifiedHTTPSConnection object at 0x7f5dfb0e5290>: Failed to establish a new connection: [Errno -4] N
on-recoverable error',))
```

I'm using requests 2.9.1 but have seen the same using latest.

I've tried turning up logging for `requests.packages.urllib3` but I'm not seeing any useful messages besides a generic: `INFO Starting new HTTPS connection (1): <hostname redacted>`

I've tried inserting a breakpoint and inspecting the `ConnectionError` but was not able to garner any useful information (though I may have missed something).

This may be my own lack of knowledge, but what library can I use to look up what errno -4 is? Google searching for a negative error code is literally impossible.

Thanks for your time!

**Comment by Lukasa** [comment]
Hrm, that's odd: on most Unices errno is restrained to a positive number. What platform are you on? Even that error message is unusual.

Does this happen consistently?

**Comment by ozdanborne** [comment]
Made it to the bottom of this. It was running in an alpine:3.3 docker container which, turns out, has a DNS resolving issue for FQDN's which wasn't patched until 3.4. So my request error was just a hostname lookup failure.

It's possible that the error was obfuscated due to the fact that my python code is packaged into a binary using pyinstaller, but I'm not sure.

Thanks for taking a look. Closing :)
